src/pages/index.js

```diff
@@ -0,0 +1,250 @@
+  .then((cards) => {
+    section = new Section(
+      {
+        items: cards,
+        renderer: (data) => {
+          const cardEl = renderCard(data);
+          section.addItem(cardEl);
+        },
+      },
+      ".gallery__cards"
+    );
+    section.renderItems();
+  })
+  .catch((error) => {
+    console.log(error);
+  });
+
+//trying something new below
+api
+  .getUserInfo()
+  .then((data) => {
+    userInfo.setUserInfo({
+      title: data.name,
+      description: data.about,
+      avatar: data.avatar,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+//something new below
+const userInfo = new UserInfo({
+  titleSelector: ".profile__title", //
+  descriptionSelector: ".profile__description", //
+  avatarSelector: ".profile__image",
+});
+
+function handleImageClick({ name, link }) {
+  popupWithImage.open(name, link);
+}
+
+function renderCard(data) {
+  const card = new Card(
+    data,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCard,
+    handleLike
+  );
+  return card.getView();
+}
+
+// something new below
+function handleProfileEditSubmit({ title, description }) {
+  editModal.setLoading(true);
+  api
+    .setUserInfo(title, description)
+    .then(() => {
+      userInfo.setUserInfo({ name: title, about: description });
+      editModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editModal.setLoading(false);
+    });
+} //edit modal allows you to update, but form reset not working
+
+function handleAddCardSubmit(name, url) {
+  addModal.setLoading(true);
+  api
+    .addCard(name, url)
+    .then((data) => {
+      const card = renderCard(data);
+      section.addItem(card);
+      addModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      addModal.setLoading(false);
+    });
+}
+
+//something new
+/*
+const newAvatar = "https://some-url.com";
+handleAvatarSubmit(newAvatar); */
+
+function handleAvatarSubmit(url) {
+  //profileAvatarPopUp.setLoading(true);
+  api
+    .updateAvatar(url)
+    .then((data) => {
+      userInfo.getAvatar(data);
+      profileAvatarPopUp.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileAvatarPopUp.setLoading(false);
+    });
+}
+
+function handleDeleteCard(cardId) {
+  cardDeletePopUp.open();
+  cardDeletePopUp.setSubmitAction(() => {
+    cardDeletePopUp.setLoading(true, "Deleting...");
+    api
+      .deleteCard(cardId)
+      .then(() => {
+        this.handleDeleteCard();
+        cardDeletePopUp.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        cardDeletePopUp.setLoading(false, "Yes");
+      });
+  });
+}
+
+function handleLike(cardId) {
+  if (cardId._isLiked) {
+    api
+      .dislikeCard(cardId._id)
+      .then(() => {
+        cardId.handleLikeIcon();
+        cardId._isLiked = false;
```

Comment:
Needs correction: please don't change the class variables directly, you should create a setter as I described in the Card class

Reply:
Still needs correction